src/dogeos/Moat.sol

```diff
@@ -0,0 +1,212 @@
+    }
+
+    // --- Setters (Owner Restricted) --- //
+
+    /**
+     * @notice Update the L2 messenger contract address.
+     * @dev Can only be called by the owner. Emits a {MessengerUpdated} event.
+     * @param _newMessenger The new L2 messenger address.
+     */
+    function updateMessenger(address _newMessenger) external onlyOwner {
+        if (_newMessenger == address(0)) {
+            revert ErrorZeroAddress();
+        }
+        address oldMessenger = messenger;
+        messenger = _newMessenger;
+        emit MessengerUpdated(oldMessenger, _newMessenger);
+    }
+
+    /**
+     * @notice Update the withdrawal fee.
+     * @dev Can only be called by the owner. Emits a {FeeUpdated} event.
+     * @param _newFee The new withdrawal fee.
+     */
+    function setFee(uint256 _newFee) external onlyOwner {
+        uint256 oldFee = withdrawalFee;
+        withdrawalFee = _newFee;
+        emit FeeUpdated(oldFee, _newFee);
+    }
+
+    /**
+     * @notice Update the minimum withdrawal amount (after fee).
+     * @dev Can only be called by the owner. Emits a {MinWithdrawalUpdated} event.
+     * @param _newMin The new minimum withdrawal amount.
+     */
+    function setMinWithdrawal(uint256 _newMin) external onlyOwner {
+        uint256 oldMin = minWithdrawalAmount;
+        minWithdrawalAmount = _newMin;
+        emit MinWithdrawalUpdated(oldMin, _newMin);
+    }
+
+    /**
+     * @notice Update the withdrawal fee recipient address.
+     * @dev Can only be called by the owner. Emits a {FeeRecipientUpdated} event.
+     * @param _newRecip The new fee recipient address.
+     */
+    function setFeeRecipient(address _newRecip) external onlyOwner {
+        if (_newRecip == address(0)) {
+            revert ErrorZeroAddress();
+        }
+        address oldRecip = feeRecipient;
+        feeRecipient = _newRecip;
+        emit FeeRecipientUpdated(oldRecip, _newRecip);
+    }
+
+    /**
+     * @notice Update the Bascule verifier contract address.
+     * @dev Can only be called by the owner. Emits a {BasculeVerifierUpdated} event.
+     * @param _newVerifier The new Bascule verifier address.
+     */
+    function setBascule(address _newVerifier) external onlyOwner {
+        // We allow setting verifier to address(0) to disable verification if needed.
+        address oldVerifier = basculeVerifier;
+        basculeVerifier = _newVerifier;
+        emit BasculeVerifierUpdated(oldVerifier, _newVerifier);
+    }
+
+    // --- Core Logic --- //
+
+    /**
+     * @notice Handles execution of a verified L1->L2 message.
+     * @dev Must be called by the designated L2 messenger. Requires message verification via Bascule.
+     * Relays the call (and value) to the target address.
+     * @param _l1Sender The original L1 sender address (passed by the messenger). Unsure if used in future depositID.
+     * @param _target The target contract address on L2.
+     * @param _data The calldata for the target contract.
```

Comment:
This shouldn't be "call data to recipient contract" it should just be bytes32 for the depostID input for Bascule